crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -51,7 +52,11 @@ fn update_old_parents(world: &mut World, parent: Entity, children: &[Entity]) {
     let mut moved: SmallVec<[HierarchyEvent; 8]> = SmallVec::with_capacity(children.len());
     for child in children {
         if let Some(previous) = update_parent(world, *child, parent) {
-            debug_assert!(parent != previous);
+            if parent == previous {
+                warn!("Entity `{child:?}` is already a child of `{parent:?}`");
```

Comment:
Probably better to just be silent about this. Success should be quiet, failure should be loud.

Reply:
🤔

Reply:
I could go either way on this 🤷

Reply:
I think I agree that this should just be silent.